x-pack/legacy/plugins/apm/public/new-platform/plugin.tsx

```diff
@@ -34,38 +37,77 @@ const App = () => {
-            <Route component={ScrollToTopOnPathChange} />
-            <LicenseProvider>
-              <Switch>
-                {routes.map((route, i) => (
-                  <Route key={i} {...route} />
-                ))}
-              </Switch>
-            </LicenseProvider>
-          </MainContainer>
-        </LoadingIndicatorProvider>
-      </UrlParamsProvider>
-    </MatchedRouteProvider>
+    <MainContainer data-test-subj="apmMainContainer">
+      <UpdateBreadcrumbs />
+      <Route component={ScrollToTopOnPathChange} />
+      <Switch>
+        {routes.map((route, i) => (
+          <Route key={i} {...route} />
+        ))}
+      </Switch>
+    </MainContainer>
   );
 };
 
-export class Plugin {
-  public start(core: LegacyCoreStart) {
-    const { i18n } = core;
+export interface ApmPluginsStart {
+  data: DataPublicPluginStart;
+}
+
+const PluginsContext = createContext({} as ApmPluginsStart);
+
+export function usePlugins() {
+  return useContext(PluginsContext);
+}
+
+export class ApmPublicPlugin
+  implements Plugin<void, void, {}, ApmPluginsStart> {
+  public setup() {}
+
+  public start(core: CoreStart, plugins: ApmPluginsStart) {
+    const i18nCore = core.i18n;
+
+    // render APM feedback link in global help menu
+    core.chrome.setHelpExtension({
+      appName: i18n.translate('xpack.apm.feedbackMenu.appName', {
+        defaultMessage: 'APM'
+      }),
+      links: [
+        {
+          linkType: 'discuss',
+          href: 'https://discuss.elastic.co/c/apm'
+        },
+        {
+          linkType: 'custom',
+          href: url.format({
+            pathname: core.http.basePath.prepend('/app/kibana'),
+            hash: '/management/elasticsearch/upgrade_assistant'
+          }),
+          content: i18n.translate('xpack.apm.helpMenu.upgradeAssistantLink', {
+            defaultMessage: 'Upgrade assistant'
+          })
+        }
+      ]
+    });
+
     ReactDOM.render(
-      <KibanaCoreContextProvider core={core}>
-        <i18n.Context>
-          <Router history={history}>
-            <LocationProvider>
-              <App />
-            </LocationProvider>
-          </Router>
-        </i18n.Context>
+      <KibanaCoreContextProvider core={core as LegacyCoreStart}>
+        <PluginsContext.Provider value={plugins}>
+          <i18nCore.Context>
+            <Router history={history}>
+              <LocationProvider>
+                <MatchedRouteProvider>
+                  <UrlParamsProvider>
+                    <LoadingIndicatorProvider>
+                      <LicenseProvider>
+                        <App />
+                      </LicenseProvider>
+                    </LoadingIndicatorProvider>
+                  </UrlParamsProvider>
+                </MatchedRouteProvider>
+              </LocationProvider>
+            </Router>
+          </i18nCore.Context>
+        </PluginsContext.Provider>
       </KibanaCoreContextProvider>,
       document.getElementById(REACT_APP_ROOT_ID)
```

Comment:
`document.getElementById(REACT_APP_ROOT_ID)` should be passed in a param `domElement` to align a bit better with the NP plugin pattern. This element reference is already in `checkForRoot`, so it can be passed as a result of that promise.

Reply:
I'm going to skip doing this for now, until we implement the `mount` method when we register the NP application.